Automattic/wp-calypso
# Signup: Reset `submitting` state when API returns `invalid_password`

Fixes https://github.com/Automattic/wp-calypso/issues/69749

Discussion in p1667488915693849-slack-C029GN3KD

## Proposed Changes

Deliberately uses `UNSAFE_componentWillReceiveProps` on the `<SignupForm>` component to set `this.setState( { submitting: false } );` when the `users/new` endpoint returns an `invalid_password` error. Notably, there's some client-side validation for the password, but `passjdks` slips past that to cause an API error.

Ideally, `<SignupForm>` wouldn't use `submitting` state that's dependent on props, but that's a refactor for another time.

### Before

![CleanShot 2022-11-03 at 10 03 39](https://user-images.githubusercontent.com/36432/199787361-acb05cdd-7bdb-4e65-9431-46f070e867ae.gif)

### After

![CleanShot 2022-11-03 at 10 00 31](https://user-images.githubusercontent.com/36432/199786824-17a7bfc7-a477-4e8f-9f8c-77faa918af04.gif)

## Testing Instructions

1. Navigate to `/start/user`.
2. Enter a email address and username that aren't associated with a WordPress.com user.
3. Enter `passjdks` as the password.
4. Submit the form.
5. Observe the form become editable again once the error displays.
